rsht.h

```diff
@@ -30,16 +30,22 @@ typedef struct rsht_ht {
   size_t *buckets;
 } rsht_ht;
 
+// returns NULL if we were unable to allocate space for buckets or entries
```

Comment:
I'd prefer it if the changes to the comments in the header file were made in a separate commit next time. Just something to keep in mind for next time, it just helps me review your changes.